src/Corax/IndexWriter.cs

```diff
@@ -84,8 +87,14 @@ private IndexWriter(IndexFieldsMapping fieldsMapping)
 
             var bufferSize = fieldsMapping!.Count;
             _buffer = new Dictionary<Slice, List<long>>[bufferSize];
+            _bufferDoubles = new Dictionary<double, List<long>>[bufferSize];
+            _bufferLongs = new Dictionary<long, List<long>>[bufferSize];
             for (int i = 0; i < bufferSize; ++i)
+            {
                 _buffer[i] = new Dictionary<Slice, List<long>>(SliceComparer.Instance);
+                _bufferDoubles[i] = new Dictionary<double, List<long>>();
+                _bufferLongs[i] = new Dictionary<long, List<long>>();
+            }
```

Comment:
we should probably add a cache here for those, no?

Reply:
Yes, but we havent applied allocations optimizations yet. We only applied for the most blatant offenders. There is lots of opportunities to improve that, and this is just one of those. I wouldnt worry much for now.